client/components/domains/map-domain-step/index.jsx

```diff
@@ -164,105 +164,21 @@ const MapDomainStep = React.createClass( {
 		this.recordEvent( 'formSubmit', this.state.searchQuery );
 		this.setState( { suggestion: null, notice: null } );
 
-		canMap( domain, error => {
-			if ( error ) {
-				this.handleValidationErrorMessage( domain, error );
-				return;
-			}
-
-			return this.props.onMapDomain( domain );
-		} );
-
 		canRegister( domain, ( error, result ) => {
 			if ( error ) {
+				if ( error.code === 'not_available_but_mappable' ) {
+					this.props.onMapDomain( domain );
+					return;
```

Comment:
Why does a function called `canRegister` calls `props.onMapDomain` if the domain is already registered?